src/clients/http.rs

```diff
@@ -110,13 +113,32 @@ impl Http {
         }
 
         // create the container with options
-        let create_result = self
-            .inner
-            .shiplift
-            .containers()
-            .create(&options_builder.build())
-            .await;
-        let id = create_result.unwrap().id;
+        let create_result = self.create_container(&options_builder).await;
+        let id = {
+            match create_result {
+                Ok(container) => container.id,
+                Err(err) => match err {
+                    shiplift::Error::Fault { code, message: _ } => {
+                        if code != StatusCode::NOT_FOUND {
```

Comment:
Can we merge this pattern matching into one and thereby reduce indentation for this logic?

I am thinking of something like:

```rust
match create_result {
	Ok(c) => c.id,
	Err(shiplift::Error::Fault { code, .. }) if code == 404 => {
		// do image pull logic
	}
	e => panic!("Failed to create container: {}", e)
}
```

Reply:
Thank you! I don't even know we can write it like that. Fixed.